src/pages/RoversPage.tsx

```diff
@@ -1,26 +1,73 @@
 import { useState, useEffect } from "react";
-import { RoverManifestDetails, getRoverManifestData } from "../api/nasaApi";
-import Rovers from "../components/Rover";
+import Rover from "../components/Rover";
 import Button from "../components/Button";
+import "./RoversPage.scss";
+import { roverCameras } from "../data/cameraData";
+import { GalleryPhotoDetails, getGalleryPhotos } from "../api/nasaApi";
+import Gallery from "../components/Gallery";
+
+const optimumSols = {
+  Curiosity: 1000,
+  Opportunity: 2000,
+  Spirit: 500,
+};
 
 const RoversPage = () => {
-  const [RoversDetails, setRoversDetails] = useState<RoverManifestDetails>();
+  const [currentRoverName, setCurrentRoverName] = useState<
+    "Curiosity" | "Opportunity" | "Spirit"
+  >("Curiosity");
```

Comment:
Define this as a type elsewhere, like:
```ts
export type RoverName = "Curiosity" | "Opportunity" | "Spirit";
```
then you can rewrite this as
```suggestion
  const [currentRoverName, setCurrentRoverName] = useState<RoverName>("Curiosity");
```

Reply:
change not completed as agreed